tst/run_tests.py

```diff
@@ -279,61 +263,85 @@ def log_init(args):
     kwargs = vars(args)
+
+    # Set MPI oversubscribe
+    if kwargs.pop("use_oversubscribe"):
+        artemis.set_mpi_oversubscribe(True)
+
+    # Check for executable path and output directory args
+    out_dir = kwargs.pop("output_dir")
+    out_dir = os.path.join(out_dir, "testing") if out_dir is not None else None
+    exe_path = kwargs.pop("exe")
+    cwd = os.getcwd()
+
     # Set the correct paths
-    artemis_exe_path = kwargs.pop("exe")
-    output_dir = kwargs.pop("output_dir")
-
-    # Check that path is valid
-    if output_dir is not None:
-        if not (os.path.exists(output_dir) and os.access(output_dir, os.W_OK)):
-            raise TestError(
-                f'Provided output directory "{output_dir}" not found or it is not writeable!'
-            )
-    # output_dir is a valid path or None
-
-    if artemis_exe_path is not None:
-        # Check that path is valid
-        if not (
-            os.path.exists(artemis_exe_path) and os.access(artemis_exe_path, os.X_OK)
-        ):
-            raise TestError(
-                f'Provided executable "{artemis_exe_path}" not found or cannot be executed!'
-            )
-
-        # If no output_dir was passed, set it to the exe path
-        if output_dir is None:
-            output_dir = os.path.dirname(artemis_exe_path)
-        # Set the valid provided executable path
-        abs_exe_path = os.path.abspath(artemis_exe_path)
-        print(f"Found local executable {abs_exe_path}")
-        print(f"Outputting results to {output_dir}")
-        artemis.set_executable(abs_exe_path, output_dir)
+    if exe_path is not None:
+        adir = os.path.join(artemis.get_artemis_dir(), "tst")
+        out_dir = os.path.join(adir, "testing") if out_dir is None else out_dir
+        reb_path = os.path.join(os.path.dirname(exe_path), "librebound.so")
+
+        if not (os.path.exists(exe_path) and os.access(exe_path, os.X_OK)):
+            raise TestError(f'Provided exe "{exe_path}" not found or cannot be run!')
+
+        if not os.path.exists(reb_path):
+            raise TestError(f'librebound.so not found at "{reb_path}"!')
+
+        abs_out_dir = os.path.abspath(out_dir)
+        abs_exe_dir = os.path.abspath(os.path.dirname(exe_path))
+        artemis.set_paths(abs_exe_dir, abs_out_dir)
+        artemis.set_supplied_exe(True)
     else:
-        # If no output_dir was passed, set it to the cwd
-        if output_dir is None:
-            output_dir = os.getcwd()
         # If we are in a directory with an executable, default to using that
-        local_path = os.path.join(os.getcwd(), "artemis")
-        if os.path.exists(local_path) and os.access(local_path, os.X_OK):
-            print(f"Found local executable {local_path}")
-            print(f"Outputting results to {output_dir}")
-            artemis.set_executable(local_path, output_dir)
+        local_path = os.path.join(cwd, "artemis")
+        if os.path.isfile(local_path) and os.access(local_path, os.X_OK):
```

Comment:
Per previous discussion I do think we should still guard this auto-find logic behind an optional `--cwd` flag, interpreted as
```python
    parser.add_argument(
        "--cwd",
        action="store_true",
        help="search local path for preexisting executable",
    )
```
And then the logic here should I think be
```python
if exe_path is not None:
  # same
else:
  use_cwd = kwargs.pop("cwd")
  if use_cwd:
    if os.path.isfile(local_path) and os.access(local_path, os.X_OK):
      # same, run code in `cwd()/testing`
    else:
      # Same, check for `CMakeCache.txt` in `cwd()`, fail if not found
  else:
    # Same, build code and run in `artemis/tst/build/src/artemis`
```

This avoids all the special cases we were discussing before about trying to run the code with default behavior and having it unintentionally pick up on a nearby `artemis` executable.